apps/mobile/src/components/Login/SignInBottomSheet.tsx

```diff
@@ -87,13 +87,46 @@ function SignInBottomSheet({ onQrCodePress, openManageWallet }: Props) {
 }
 
 function FarcasterBottomSheetRow() {
-  const { open: handleConnectFarcaster } = useLoginWithFarcaster();
+  const [isFarcasterLoading, setIsFarcasterLoading] = useState(false);
+  const [attemptReconnect, setAttemptReconnect] = useState(false);
+
+  const {
+    open: handleConnectFarcaster,
+    isSuccess,
+    isConnected,
+    isPolling,
+  } = useLoginWithFarcaster({
+    setIsFarcasterLoading,
+  });
+
+  const [appState, setAppState] = useState(AppState.currentState);
+
+  useEffect(() => {
+    const subscription = AppState.addEventListener('change', (nextAppState) => {
+      if (appState.match(/inactive|background/) && nextAppState === 'active') {
+        if (isPolling || (!isConnected && !isSuccess)) {
+          setAttemptReconnect(true);
+        }
+        setIsFarcasterLoading(false);
+      }
+      setAppState(nextAppState);
+    });
+
+    return () => {
+      subscription.remove();
+    };
+  }, [appState, isPolling, isConnected, isSuccess]);
+
+  const handleConnectFarcasterPress = useCallback(() => {
+    handleConnectFarcaster(attemptReconnect);
+  }, [attemptReconnect, handleConnectFarcaster]);
```

Comment:
so the only thing i'm concerned about with this PR is farcaster-related-logic being split in this component (`FarcasterBottomSheetRow`) as well as the main hook `useLoginWithFarcaster`

from what i can tell, the `appState / useState` and `useEffect` can both live inside `useLoginWithFarcaster`. the only variable used by this component is `isFarcasterLoading`, but even that i think can be a state that's defined inside `useLoginWithFarcaster` and exported out for this component to use